src/WeihanLi.Npoi/NpoiExtensions.cs

```diff
@@ -158,29 +173,59 @@ public static DataTable ToDataTable(this ISheet sheet, int headerRowIndex)
 
                 if (row.RowNum == headerRowIndex)
                 {
-                    foreach (var cell in row)
-                    {
-                        if (cell is null)
-                        {
-                            continue;
-                        }
-                        dataTable.Columns.Add(cell.GetCellValue(typeof(string), formulaEvaluator)?.ToString().Trim());
-                    }
+                    LoadHeader(formulaEvaluator, dataTable, row, maxColumns);
                 }
                 else
                 {
-                    var dataRow = dataTable.NewRow();
+                    LoadRow(formulaEvaluator, dataTable, row, removeEmptyRows, maxColumns);
+                }
+            }
+
+            return dataTable;
 
-                    foreach (var cell in row)
+            static void LoadHeader(IFormulaEvaluator formulaEvaluator, DataTable dataTable, IRow row, int? maxColumns)
+            {
+                foreach (var cell in row)
+                {
+                    if (cell is null)
                     {
-                        dataRow[cell.ColumnIndex] = cell.GetCellValue(typeof(string), formulaEvaluator);
+                        continue;
```

Comment:
The default value would be `null` or `DBNull` if continue?
I'm not sure about this, would have a test on my side.